### PR TITLE
Using helper functions for clCreateKernel

### PR DESCRIPTION
**Pull request by ellnor01**
Uses of clCreateKernel following create program helper
functions, have been incorporated into
create_single_kernel_helper when suitable.

Contributes #31

Signed-off-by: Ellen Norris-Thompson <ellen.norris-thompson@arm.com>

### CODE REVIEW

File: test_conformance/math_brute_force/main.cpp [hunk 1]
```diff
@@ -1392,17 +1392,14 @@ int InitILogbConstants( void )
...
-    error = create_single_kernel_helper(gContext, &query, NULL, 1, &kernel, NULL);
-    if (NULL == query || error)
-    {
-        vlog_error( "Error: Unable to create program to get FP_ILOGB0 and FP_ILOGBNAN for the device. (%d)", error );
-        return error;
-    }
-
-    cl_kernel k = clCreateKernel( query, "GetILogBConstants", &error );
-    if( NULL == k || error)
+    cl_kernel k;
```
Comment (jrprice) [inline test_conformance/math_brute_force/main.cpp:1395]: Prefer `clKernelWrapper`.

Could also switch the to program to `clProgramWrapper`, but that's getting further from the scope of this PR so fine to leave that for later.

File: test_conformance/math_brute_force/main.cpp [hunk 2]
```diff
@@ -1447,16 +1444,13 @@ int IsTininessDetectedBeforeRounding( void )
     "}\n";
 
     cl_program query;
-    error = create_single_kernel_helper(gContext, &query, NULL, 1, &kernel, NULL);
+    cl_kernel k;
```
Comment (jrprice) [inline test_conformance/math_brute_force/main.cpp:1447]: `clKernelWrapper`

File: test_conformance/math_brute_force/main.cpp [hunk 3]
```diff
@@ -1590,16 +1573,13 @@ static int IsInRTZMode( void )
...
```
Comment (jrprice) [inline test_conformance/math_brute_force/main.cpp:1576]: `clKernelWrapper`

**Review by jrprice — APPROVED** [review]
LGTM, just suggest that we switch to the managed `*Wrapper` classes at the same time as changing code like this.

Appveyor Windows build is failing.

**Comment by kpet** [comment]
@ellnor01 Could you take a look at the Windows CI and possibly use wrappers where suggested?

**Review by jrprice — DISMISSED** [review]

File: test_common/harness/imageHelpers.cpp [hunk 4]
```diff
@@ -2953,64 +2954,41 @@ int DetectFloatToHalfRoundingMode(
...
+        err = clSetKernelArg(kernel, 0, sizeof(cl_mem), &inBuf);
         if (err)
         {
             log_error("Error: could not set argument 0 of kernel in "
                       "DetectFloatToHalfRoundingMode (%d)",
                       err);
             clReleaseMemObject(inBuf);
             clReleaseMemObject(outImage);
-            clReleaseKernel(k);
+            clReleaseKernel(kernel);
```
Comment (kpet) [inline test_common/harness/imageHelpers.cpp:2965]: `kernel` is now a wrapper. You should remove the calls to `clReleaseKernel` altogether. There are a few more occurrences below.

**Review by kpet — CHANGES_REQUESTED** [review]

**Review by kpet — APPROVED** [review]
LGTM

**Pull request merged**